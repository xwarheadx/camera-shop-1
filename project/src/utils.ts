import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


