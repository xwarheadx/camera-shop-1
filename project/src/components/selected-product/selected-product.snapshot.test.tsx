import { productMock } from '../../test/test-mocks';
import { renderWithProviders } from '../../test/utils/render-with-redux';
import SelectedProduct from './selected-product';

test('Render SelectedProduct', () => {
  const { asFragment } = renderWithProviders(<SelectedProduct camera={productMock}/>);
  expect(asFragment()).toMatchSnapshot();
});
