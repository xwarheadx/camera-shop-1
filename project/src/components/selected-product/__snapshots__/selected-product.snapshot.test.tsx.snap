// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SelectedProduct 1`] = `
<DocumentFragment>
  <div
    class="page-content__section"
  >
    <section
      class="product"
    >
      <div
        class="container"
      >
        <div
          class="product__img"
        >
          <picture>
            <source
              srcset="../img/content/look-identify.webp, ../img/content/look-identify@2x.webp 2x"
              type="image/webp"
            />
            <img
              alt="Ретрокамера Das Auge IV"
              height="480"
              src="../img/content/look-identify.jpg"
              srcset="../img/content/look-identify@2x.jpg 2x"
              width="560"
            />
          </picture>
        </div>
        <div
          class="product__content"
        >
          <h1
            class="title title--h3"
          >
            Look Identify
          </h1>
          <div
            class="rate product__rate"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                data-testid="star-test"
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                data-testid="star-test"
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                data-testid="star-test"
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                data-testid="star-test"
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                data-testid="star-test"
                xlink:href="#icon-full-star"
              />
            </svg>
            <p
              class="visually-hidden"
            >
              Рейтинг: 5
            </p>
            <p
              class="rate__count"
            >
              <span
                class="visually-hidden"
              >
                Всего оценок:
              </span>
              18
            </p>
          </div>
          <p
            class="product__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            12 6000 ₽
          </p>
          <button
            class="btn btn--purple"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="24"
            >
              <use
                xlink:href="#icon-add-basket"
              />
            </svg>
            Добавить в корзину
          </button>
          <div
            class="tabs product__tabs"
          >
            <div
              class="tabs__controls product__tabs-controls"
            >
              <button
                class="tabs__control"
                data-testid="characteristics-button-test"
                type="button"
              >
                Характеристики
              </button>
              <button
                class="tabs__control is-active"
                data-testid="info-button-test"
                type="button"
              >
                Описание
              </button>
            </div>
            <div
              class="tabs__content"
            >
              <div
                class="tabs__element"
              >
                <ul
                  class="product__tabs-list"
                >
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Артикул:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      LD2000
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Категория:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Фотоаппарат
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Тип камеры:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Коллекционная
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Уровень:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Любительский
                    </p>
                  </li>
                </ul>
              </div>
              <div
                class="tabs__element is-active"
              >
                <div
                  class="product__tabs-text"
                >
                  <p>
                    Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники.
                  </p>
                  <p>
                    Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
